packages/@react-spectrum/list/test/ListViewDnd.test.js

```diff
@@ -335,6 +342,196 @@ describe('ListView', function () {
+
+        let grid1 = getAllByRole('grid')[0];
+        let grid2 = getAllByRole('grid')[1];
+        let list1rows = within(grid1).getAllByRole('row');
+        let list2rows = within(grid2).getAllByRole('row');
+        let cell = within(list1rows[0]).getByRole('gridcell');
+
+        expect(within(list1rows[0]).getByRole('gridcell')).toHaveTextContent('Item One');
+        expect(within(list1rows[1]).getByRole('gridcell')).toHaveTextContent('Item Two');
+        expect(within(list1rows[2]).getByRole('gridcell')).toHaveTextContent('Item Three');
+
+        expect(within(list2rows[0]).getByRole('gridcell')).toHaveTextContent('Item Seven');
+        expect(within(list2rows[1]).getByRole('gridcell')).toHaveTextContent('Item Eight');
+        expect(within(list2rows[2]).getByRole('gridcell')).toHaveTextContent('Item Nine');
+
+        act(() => userEvent.click(within(list1rows[0]).getByRole('checkbox')));
+        act(() => userEvent.click(within(list1rows[2]).getByRole('checkbox')));
+
+        let dataTransfer = new DataTransfer();
+        fireEvent.pointerDown(cell, {pointerType: 'mouse', button: 0, pointerId: 1, clientX: 0, clientY: 0});
+        fireEvent(cell, new DragEvent('dragstart', {dataTransfer, clientX: 0, clientY: 0}));
+        expect(onDragStart).toHaveBeenCalledTimes(1);
+
+        fireEvent.pointerMove(cell, {pointerType: 'mouse', button: 0, pointerId: 1, clientX: 200, clientY: -1});
+        fireEvent(cell, new DragEvent('drag', {dataTransfer, clientX: 200, clientY: -1}));
+        fireEvent(grid2, new DragEvent('dragover', {dataTransfer, clientX: 200, clientY: -1}));
+        fireEvent.pointerUp(cell, {pointerType: 'mouse', button: 0, pointerId: 200, clientX: 1, clientY: -1});
+
+        fireEvent(cell, new DragEvent('dragend', {dataTransfer, clientX: 200, clientY: -1}));
+        expect(onDragEnd).toHaveBeenCalledTimes(1);
+
+        fireEvent(grid2, new DragEvent('drop', {dataTransfer, clientX: 200, clientY: -1}));
+        act(() => jest.runAllTimers());
+        await act(async () => Promise.resolve());
+        expect(onDrop).toHaveBeenCalledTimes(1);
+
+        act(() => jest.runAllTimers());
+
+        grid1 = getAllByRole('grid')[0];
+        grid2 = getAllByRole('grid')[1];
+        list1rows = within(grid1).getAllByRole('row');
+        list2rows = within(grid2).getAllByRole('row');
+
+        expect(within(list1rows[0]).getByRole('gridcell')).toHaveTextContent('Item Two');
+        expect(within(list1rows[1]).getByRole('gridcell')).toHaveTextContent('Item Four');
+        expect(within(list1rows[2]).getByRole('gridcell')).toHaveTextContent('Item Five');
+
+        expect(within(list2rows[0]).getByRole('gridcell')).toHaveTextContent('Item One');
+        expect(within(list2rows[1]).getByRole('gridcell')).toHaveTextContent('Item Three');
+        expect(within(list2rows[2]).getByRole('gridcell')).toHaveTextContent('Item Seven');
+
+        // expect(document.activeElement).toBe(list2rows[0]);
```

Comment:
For some reason, the first list is what's focused here in the test

Reply:
where is item 6?

Reply:
I was just checking the first three items in each list. I suppose I could check every item in both lists, but since we were just dragging a couple to the start of the other, I did that

Reply:
I'll add a check for all to be explicit, and a check for length

Reply:
I think I expected it to be before item 7, but it didn't look like there was room for it, that's why I was asking